src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,331 @@
+                <Text style={styles.label}>{t('swapReviewScreen.estimatedGas')}</Text>
+              </View>
+              <View style={[styles.column, styles.tokenDisplayView]}>
+                <TokenDisplay
+                  style={styles.transactionDetailsRightText}
+                  amount={divideByWei(swapInfo?.unvalidatedSwapTransaction?.gas)}
+                  tokenAddress={fromToken}
+                  showLocalAmount={false}
+                  testID={'EstimatedGas'}
+                />
+              </View>
+            </View>
+          </View>
+        </ScrollView>
+      )}
+      <Button
+        style={{ padding: Spacing.Regular16 }}
+        onPress={() => Logger.debug('TODO Perform Swap!!!')}
+        text={t('swapReviewScreen.complete')}
+        size={BtnSizes.FULL}
+        disabled={loading}
+      />
+      {/** Estimated amount dialog */}
+      <Dialog
+        title={t('swapReviewScreen.estimatedAmountTitle')}
+        isVisible={estimatedModalVisible}
+        actionText={t('dismiss')}
+        actionPress={() => setEstimatedDialogVisible(false)}
+        isActionHighlighted={false}
+        onBackgroundPress={() => setEstimatedDialogVisible(false)}
+      >
+        {t('swapReviewScreen.estimatedAmountBody', {
+          slippagePercent: maxSlippagePercent,
+        })}
+      </Dialog>
+    </SafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  safeAreaContainer: {
+    flex: 1,
+  },
+  loadingContentContainer: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  contentContainer: {
+    padding: Spacing.Regular16,
+    flexGrow: 1,
+  },
+  subContentContainer: {
+    paddingBottom: Spacing.Regular16,
+  },
+  tallRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Regular16,
+  },
+  row: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    paddingBottom: Spacing.Smallest8,
+  },
+  // Used for large text display
+  column: {
+    maxWidth: '50%',
+  },
+  tokenDisplayView: {
+    justifyContent: 'space-between',
+  },
+  separator: {
+    height: 1,
+    width: '100%',
+    backgroundColor: colors.gray2,
+  },
+  // No shared font styles for this label - done manually
+  transactionDetailsRightText: {
+    alignSelf: 'flex-end',
+    textAlign: 'right',
+    textAlignVertical: 'center',
+    fontWeight: '400',
+    fontSize: 16,
+    lineHeight: 24,
+    fontFamily: 'Inter',
+    fontStyle: 'normal',
+  },
+  touchableRow: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  transactionDetailsLabel: {
+    ...fontStyles.label,
+    paddingBottom: Spacing.Smallest8,
+    color: colors.gray4,
+  },
+  label: {
+    ...fontStyles.regular,
+  },
+  tokenText: {
+    alignSelf: 'flex-end',
+    textAlign: 'right',
+    ...fontStyles.large,
+  },
+  tokenSubText: {
+    alignSelf: 'flex-end',
```

Comment:
i don't think you need `alignSelf` in this or the `tokenText` styles

Reply:
Styles adjusted in https://github.com/valora-inc/wallet/pull/2767/commits/70960e345f0debe2cbefb3fce5e70ef19e3fddca!